### ISSUE TITLE
Two ports per host

### ISSUE THREAD

**Issue by TeddyBatlink** [NginxProxyManager/nginx-proxy-manager#4004]
Hello. I have a need to pass both ports (80 and 443) to my server, but I can't find a solution exactly how to do this. Similar questions have already been found on the Internet, but there are no answers there. There is no such function in the NPM interface, I think this can be done in the NPM configuration files, or in the "advanced" tab, but I don't know how. Can you help with this issue? Any option will suit me, even if I need to manually configure nginx, as long as this option works.

**Comment by TeddyBatlink** [comment]
> What is your current "docker run" command or compose file or equivalent method to start NPM?

NPM is launched using the docker-compose file.yml with these parameters:

```
version: '3.8'
services:
app:
image: 'jc21/nginx-proxy-manager:latest'
restart: unless-stopped
ports:
- '80:80'
- '81:81'
- '443:443'
volumes:
- ./data:/data
- ./letsencrypt:/etc/letsencrypt
```

**Comment by shalafi99** [comment]
Apologies, I think I misread your original question - you would like a single proxy host entry in NPM to handle both http and https for a given server?

**Comment by TeddyBatlink** [comment]
> Apologies, I think I misread your original question - you would like a single proxy host entry in NPM to handle both http and https for a given server?

Yes, that's right.

**Comment by shalafi99** [comment]
I did some testing and got it working using the advanced config shown below:

```
location / {
    # backend targets
    set $backend_http "http://test:80";
    set $backend_https "https://test:443";
    
    # proxy to https backend
    if ($scheme = "https") {
        proxy_pass $backend_https;
    }
    
    # proxy to http backend
    if ($scheme = "http") {
        proxy_pass $backend_http;
    }
}
```

where "test" is the name of the nginx container I created to sit behind NPM.

tests using curl to observe the desired behavior, since the web browser was adamant on upgrading http to https no matter what I tried:
_(I redacted the values for my environment with mydomain and myip)_

```
C:\>curl -f http://mydomain -v
* Host mydomain:80 was resolved.
* IPv6: (none)
* IPv4: myip
*   Trying myip:80...
* Connected to mydomain (myip) port 80
> GET / HTTP/1.1
> Host: mydomain
> User-Agent: curl/8.7.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: openresty
< Date: Thu, 19 Sep 2024 01:45:16 GMT
< Content-Type: text/plain
< Content-Length: 16
< Connection: keep-alive
<
Hello from HTTP!* Request completely sent off
* Connection #0 to host mydomain left intact
```


```
C:\>curl -f https://mydomain -v
* Host mydomain:443 was resolved.
* IPv6: (none)
* IPv4: myip
*   Trying myip:443...
* Connected to mydomain (myip) port 443
* schannel: disabled automatic use of client certificate
* ALPN: curl offers http/1.1
* ALPN: server accepted http/1.1
* using HTTP/1.x
> GET / HTTP/1.1
> Host: mydomain
> User-Agent: curl/8.7.1
> Accept: */*
>
* Request completely sent off
< HTTP/1.1 200 OK
< Server: openresty
< Date: Thu, 19 Sep 2024 01:48:03 GMT
< Content-Type: text/plain
< Content-Length: 17
< Connection: keep-alive
<
Hello from HTTPS!* Connection #0 to host mydomain left intact
```

nginx.conf file for this test:

```
events {}

http {
    server {
        listen 80;
        server_name mydomain;

        location / {
            return 200 "Hello from HTTP!";
        }
    }

    server {
        listen 443 ssl;
        server_name mydomain;

        ssl_certificate /etc/nginx/ssl/test.pem;
        ssl_certificate_key /etc/nginx/ssl/test.key;

        location / {
            return 200 "Hello from HTTPS!";
        }
    }
}
```

Although it looks weird to have a proxy host which completely ignores what you configure in the "scheme" and "forward port" fields. 

![image](https://github.com/user-attachments/assets/279d6191-100b-4bd5-aade-77783b494412)

I would rather configure two proxy hosts, one for each port, but NPM does not allow it with the same FQDN 😏 (which must be the reason for your post here).

Cheers

**Comment by TeddyBatlink** [comment]
> I did some testing and got it working using the advanced config shown below:
> 
> ```
> location / {
>     # backend targets
>     set $backend_http "http://test:80";
>     set $backend_https "https://test:443";
>     
>     # proxy to https backend
>     if ($scheme = "https") {
>         proxy_pass $backend_https;
>     }
>     
>     # proxy to http backend
>     if ($scheme = "http") {
>         proxy_pass $backend_http;
>     }
> }
> ```

I understand that this code should be written in the "advanced" tab? If so, will that be enough? Or will it be necessary to additionally write code for the nginx.conf file?

**Comment by shalafi99** [comment]
> > I did some testing and got it working using the advanced config shown below:
> > ```
> > location / {
> >     # backend targets
> >     set $backend_http "http://test:80";
> >     set $backend_https "https://test:443";
> >     
> >     # proxy to https backend
> >     if ($scheme = "https") {
> >         proxy_pass $backend_https;
> >     }
> >     
> >     # proxy to http backend
> >     if ($scheme = "http") {
> >         proxy_pass $backend_http;
> >     }
> > }
> > ```
> 
> I understand that this code should be written in the "advanced" tab? If so, will that be enough? Or will it be necessary to additionally write code for the nginx.conf file?

Yes, this section above goes into the Advanced tab for the NPM proxy host.

The code for nginx.conf is not necessary for NPM, I just included it here for completeness (how I tested what I provided) and for reference, in case you need to troubleshoot and compare the test backend I used with your actual setup.

**Comment by janekska** [comment]
Hi, 
It looks like it is exactly that I need too.

But my reverse proxy behaves diferently and I am getting some certificate error:
```
curl -f https://my.doma.in -v
*   Trying XX.XXX.XXX.26:443...
* Connected to my.doma.in (XX.XXX.XXX.26) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS header, Unknown (21):
* TLSv1.3 (IN), TLS alert, unrecognized name (624):
* error:0A000458:SSL routines::tlsv1 unrecognized name
* Closing connection 0
curl: (35) error:0A000458:SSL routines::tlsv1 unrecognized name
```

While I still not sure are this problem came from my local machine.
It is VM with mailinabox script, that I ran and I am not sure that it managed to make right certificate.
While I can test it working on lan https via browser.
Browser tells that domain is right, but there is the same date for certificate start and end.

brgds
